sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/SimpleHttpHeartbeatSender.java

```diff
@@ -108,15 +110,7 @@ private InetSocketAddress getAvailableAddress() {
                 if (ipPortStr.trim().length() == 0) {
                     continue;
                 }
-                if (ipPortStr.startsWith("http://")) {
-                    ipPortStr = ipPortStr.trim().substring(7);
-                }
-                String[] ipPort = ipPortStr.trim().split(":");
-                int port = 80;
-                if (ipPort.length > 1) {
-                    port = Integer.parseInt(ipPort[1].trim());
-                }
-                newAddrs.add(new InetSocketAddress(ipPort[0].trim(), port));
+                newAddrs.add(new DashboardConfig(ipPortStr));
```

Comment:
As described in https://github.com/alibaba/Sentinel/issues/332#issuecomment-450089404, we need a property item (like `-Dcsp.sentinel.heartbeat.api.path`)  to configure the API path of the heartbeat endpoint. It's expected to be the entire API path rather than the context prefix.

Reply:
It might be better use one property to set schema/host/port/path etc. It's friendly for user to set up, and don't need to change `org.springframework.cloud:spring-cloud-starter-alibaba-sentinel` for this feature.